### PR TITLE
[dv] Add fusesoc generator for CSR assert module

### PR DESCRIPTION
**Pull request by sriyerg**
THis is a draft PR that adds a fusesoc generator to create the CSR
assert module that was primarily developed for FPV, so that it can be
used in DV sims as well.

Currently there are issues in the generated CSR assert module causing
compile to fail.

Signed-off-by: Srikrishna Iyer <sriyer@google.com>

### CODE REVIEW

**Comment by cindychip** [comment]
Thanks Sri for supporting to generate FPV csr checks on the fly. I am working on fixing the compile errors from csr_assertions so the CI in this PR will pass.

File: hw/formal/tools/csr_assert_gen/csr_assert_gen.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                'files': [
+                    csr_assert_file,
+                ],
+                'file_type': 'systemVerilogSource'
+            },
+        },
+        'targets': {
+            'default': {
+                'filesets': [
+                    'files_rtl',
```
[the remark this answers is not in the document]
Comment (sriyerg) [reply hw/formal/tools/csr_assert_gen/csr_assert_gen.py]: Good catch, fixed!

File: hw/ip/uart/dv/bind/uart_bind.core [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# SPDX-License-Identifier: Apache-2.0
+name: "lowrisc:dv:uart_bind:0.1"
+description: "Bind file for binding asserts and covers to UART & its sub-modules."
+filesets:
+  files_dv:
+    depend:
+      - lowrisc:ip:uart
+      - lowrisc:tlul:common
+      - lowrisc:fpv:csr_assert_gen
+      - lowrisc:fpv:uart_csr_assert
```
[the remark this answers is not in the document]
Comment (sriyerg) [reply hw/ip/uart/dv/bind/uart_bind.core]: Agreed! Removed.

**Comment by sriyerg** [comment]
For the following IPs the CSr assert module is throwing compile errors: flash_ctrl, alert_handler, spidev, usbdev
There may be fixes needed in the template.

**Pull request merged**